### ISSUE TITLE
Background is white in dark mode

### ISSUE THREAD

**Issue by CorvetteCole** [gaphor/gaphor#2018]
### Describe the bug
On Linux, version 2.15 installed through Flathub. The application properly detects dark mode and the entire UI is in dark mode except for the background

### To Reproduce
1. Install from Flathub on Linx
2. Open and create new model
3. Enable dark mode
4. Observe white background

Gaphor version:    2.15.0
Operating System:  Linux (6.1.6-arch1-3)
Python version:    3.10.6
GTK version:       4.8.3
Adwaita version:   1.2.1
PyGObject version: 3.42.2
Pycairo version:   1.23.0
Cairo version:     1.16.0
Pango version:     1.50.12
Display:           GdkWaylandDisplay

No errors have been intercepted. We’re good.

### Expected behavior
The background to be black, as shown in screenshots online

### Screenshots
![image](https://user-images.githubusercontent.com/19213965/213592332-a0455ee8-6372-40ee-bf27-d61c50ac5cd6.png)

### OS
-  Arch Linux

### Version
Version of Gaphor: 2.15

**Comment by danyeaw** [comment]
Hi @CorvetteCole, instructions on how to get a dark background with white text are here in the docs:
https://gaphor.readthedocs.io/en/latest/style_sheets.html

**Comment by CorvetteCole** [comment]
Interesting @danyeaw , you're totally right I missed that. This is still pretty darn counter-intuitive though, especially since if you make a new graph it will be light again, and since the rest of the UI switches automatically. Is there a case for some sort of change here?

**Comment by CorvetteCole** [comment]
(I understand why that would be very difficult given the CSS styling built-in already)

**Comment by amolenaar** [comment]
We would basically need a different style sheet for dark mode. The style sheet is also used when you export the diagram as an image. So it seemed reasonable to keep the drawing area off-white for now.

An alternative would be a `@media` query in the CSS :wink:.

**Comment by danyeaw** [comment]
What if we introduce a default color palette depending on light/dark mode?

Then the CSS only overrides the default. For example, in dark mode, the background of the diagram is always black, unless you override the CSS attribute for the background of the diagram.

When exporting images, it could default to the light pallette, and maybe could be configurable to export dark as well.

**Comment by CorvetteCole** [comment]
I like that idea personally, it feels more intuitive from a new user perspective

**Comment by amolenaar** [comment]
Yeah, I was thinking that this would go very well with [variables in CSS](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties).

@CorvetteCole You you want to take a look at this? The styling code is located in [gaphor.core.styling](https://github.com/gaphor/gaphor/tree/main/gaphor/core/styling). It's using [tinycss2](https://pypi.org/project/tinycss2/).

**Comment by amolenaar** [comment]
@CorvetteCole @danyeaw :smile: 

[Screencast from 2023-01-24 22-50-53.webm](https://user-images.githubusercontent.com/96249/214429330-c2e5ce40-761f-43b4-82ad-7673c7c94c58.webm)

The CSS can be written like this:

```css
* {
    background-color: transparent;
}

diagram {
    background-color: white;
}

@media dark-mode {
    * {
        color: white;
    }
    diagram {
        background-color: #1e1e1e;
    }
}
```

**Comment by CorvetteCole** [comment]
that.. is... BEAUTIFUL! Exactly what I was picturing would happen

**Comment by danyeaw** [comment]
Closed by #2033
